hathor/transaction/storage/transaction_storage.py

```diff
@@ -674,13 +773,21 @@ def _save_transaction(self, tx: BaseTransaction, *, only_metadata: bool = False)
     def _build_indexes_manager(self) -> IndexesManager:
         return MemoryIndexesManager()
 
-    def _reset_cache(self) -> None:
+    def reset_indexes(self) -> None:
+        assert self.indexes is not None
+        self.indexes.force_clear_all()
+        self._reset_cache(build_indexes=False)
+
+    def _reset_cache(self, *, build_indexes: bool = True) -> None:
```

Comment:
Should we reset here `_latest_timestamp` and `_first_timestamp` as well that you also set in the `_update_caches` method?

Reply:
I don't get what you mean,`_reset_cache` does in fact reset `_latest_timestamp` and `_first_timestamp`.